Running a bundle with dependencies from a tag fails

## Describe the bug
Dependencies assume that is has access to the porter manifest but it's only available when running the bundle from a file (i.e. when you are in dev mode and running it from the local file system). When running a bundle from a tag, the porter client doesn't have acess to the porter.yaml.

For now I'm going to turn off the piece of code that is trying to access the manifest to avoid triggering the bug. I think it would be incredibly useful to porter to know when it's running a porter bundle and be able to use what's in the manifest though. That should be a separate feature.

## To Reproduce

1. Install a bundle from a tag with a dependency


## Porter Command and Output
```
$ porter install porter-wordpress --tag deislabs/porter-wordpress-bundle:v0.1.1 --param namespace=test --param wordpress-password=passw0rd --param mysql#mysql-user=wordpress -c minikube --debug 

Resolved porter binary from /usr/local/bin/porter to /Users/carolynvs/.porter/porter
Resolved dependency mysql to deislabs/porter-mysql-bundle:v0.1.1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x110 pc=0x212676c]

goroutine 1 [running]:
github.com/deislabs/porter/pkg/porter.(*dependencyExecutioner).prepareDependency(0xc0005a7aa8, 0xc0000c8bd0, 0x0, 0x20)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/pkg/porter/dependencies.go:193 +0x2fc
github.com/deislabs/porter/pkg/porter.(*dependencyExecutioner).Prepare(0xc0005a7aa8, 0x0, 0x0, 0xc000042640, 0x50, 0xc00009ca80, 0x5c, 0x7ffeefbff6cf, 0x10, 0x1, ...)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/pkg/porter/dependencies.go:70 +0xd1
github.com/deislabs/porter/pkg/porter.(*Porter).InstallBundle(0xc000332400, 0x0, 0x0, 0xc000042640, 0x50, 0xc00009ca80, 0x5c, 0x7ffeefbff6cf, 0x10, 0x1, ...)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/pkg/porter/install.go:36 +0x239
main.buildBundleInstallCommand.func2(0xc0004a7680, 0xc000332a80, 0x1, 0x8, 0x0, 0x0)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/cmd/porter/bundle.go:85 +0x54
github.com/deislabs/porter/vendor/github.com/spf13/cobra.(*Command).execute(0xc0004a7680, 0xc000332a00, 0x8, 0x8, 0xc0004a7680, 0xc000332a00)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/deislabs/porter/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00018a000, 0xc00018b900, 0xc00018b180, 0xc00018af00)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/deislabs/porter/vendor/github.com/spf13/cobra.(*Command).Execute(0xc00018a000, 0xc0005a7f88, 0xc00009c058)
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
	/home/vsts/work/1/s/gopath/src/github.com/deislabs/porter/cmd/porter/main.go:18 +0x27
```

## Version
This affects both latest and canary.

```
porter v0.20.2-beta.1-16-ge225633d (e225633d)
```

--- a/docs/content/dependencies.md
+++ b/docs/content/dependencies.md
@@ -41,6 +41,9 @@ dependencies:
       mysql_user: wordpress
 ```
 
+There is a bug with defaulting parameters from the manifest right now so it's turned off but we plan on adding that back soon.
+Watch [#800](https://github.com/deislabs/porter/issues/800) for when it's available again.
+
 ## Specifying parameters
 
 You can specifying parameters for a dependent bundle on the command-line using the following syntax

--- a/pkg/porter/dependencies.go
+++ b/pkg/porter/dependencies.go
@@ -17,7 +17,8 @@ import (
 
 type dependencyExecutioner struct {
 	*context.Context
-	Manifest        *manifest.Manifest
+	// See https://github.com/deislabs/porter/issues/799
+	// Manifest        *manifest.Manifest
 	Resolver        BundleResolver
 	CNAB            CNABProvider
 	InstanceStorage instancestorage.StorageProvider
@@ -34,8 +35,9 @@ func newDependencyExecutioner(p *Porter) *dependencyExecutioner {
 		Registry: p.Registry,
 	}
 	return &dependencyExecutioner{
-		Context:         p.Context,
-		Manifest:        p.Manifest,
+		Context: p.Context,
+		// See https://github.com/deislabs/porter/issues/799
+		// Manifest:        p.Manifest,
 		Resolver:        resolver,
 		CNAB:            p.CNAB,
 		InstanceStorage: p.InstanceStorage,
@@ -190,6 +192,8 @@ func (e *dependencyExecutioner) prepareDependency(dep *queuedDependency) error {
 	//   DEP:
 	//     parameters:
 	//       PARAM: VALUE
+	/* Disabling this until we have access to the manifest when working with a bundle from a tag
+	   See https://github.com/deislabs/porter/issues/799
 	for paramName, value := range e.Manifest.Dependencies[dep.Alias].Parameters {
 		// Make sure the parameter is defined in the bundle
 		if _, ok := depParams[paramName]; !ok {
@@ -201,6 +205,7 @@ func (e *dependencyExecutioner) prepareDependency(dep *queuedDependency) error {
 		}
 		dep.Parameters[paramName] = value
 	}
+	*/
 
 	// Handle any parameter overrides for the dependency defined on the command line
 	// --param DEP#PARAM=VALUE

--- a/tests/dependencies_test.go
+++ b/tests/dependencies_test.go
@@ -65,6 +65,8 @@ func installWordpressBundle(p *porter.TestPorter) (namespace string) {
 		"namespace=" + namespace,
 		"wordpress-name=porter-ci-wordpress-" + namespace,
 		"mysql#mysql-name=porter-ci-mysql-" + namespace,
+		// TODO: Remove this once we put back in dependency parameter defaults in #800
+		"mysql#mysql-user=wordpress",
 	}
 	err := installOpts.Validate([]string{}, p.Context)
 	require.NoError(p.T(), err, "validation of install opts for root bundle failed")
@@ -113,6 +115,8 @@ func upgradeWordpressBundle(p *porter.TestPorter, namespace string) {
 		"namespace=" + namespace,
 		"wordpress-name=porter-ci-wordpress-" + namespace,
 		"mysql#mysql-name=porter-ci-mysql-" + namespace,
+		// TODO: Remove this once we put back in dependency parameter defaults in #800
+		"mysql#mysql-user=wordpress",
 	}
 	err := upgradeOpts.Validate([]string{}, p.Context)
 	require.NoError(p.T(), err, "validation of upgrade opts for root bundle failed")
@@ -141,6 +145,8 @@ func invokeWordpressBundle(p *porter.TestPorter, namespace string) {
 		"namespace=" + namespace,
 		"wordpress-name=porter-ci-wordpress-" + namespace,
 		"mysql#mysql-name=porter-ci-mysql-" + namespace,
+		// TODO: Remove this once we put back in dependency parameter defaults in #800
+		"mysql#mysql-user=wordpress",
 	}
 	err := invokeOpts.Validate([]string{}, p.Context)
 	require.NoError(p.T(), err, "validation of invoke opts for root bundle failed")
@@ -169,6 +175,8 @@ func uninstallWordpressBundle(p *porter.TestPorter, namespace string) {
 		"namespace=" + namespace,
 		"wordpress-name=porter-ci-wordpress-" + namespace,
 		"mysql#mysql-name=porter-ci-mysql-" + namespace,
+		// TODO: Remove this once we put back in dependency parameter defaults in #800
+		"mysql#mysql-user=wordpress",
 	}
 	err := uninstallOptions.Validate([]string{}, p.Context)
 	require.NoError(p.T(), err, "validation of uninstall opts for root bundle failed")

